engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/text/util/TypeToDisplayTextNode.java

```diff
@@ -71,6 +72,12 @@ String doDisplay(
       } catch (UnsupportedMessageException e) {
         throw new IllegalStateException("Receiver declares a meta object, but does not return it.");
       }
+    } else if (exceptions.hasExceptionMessage(value)) {
+      try {
+        return strings.asString(exceptions.getExceptionMessage(value));
+      } catch (UnsupportedMessageException e) {
+        throw new IllegalStateException(e);
+      }
```

Comment:
This looks unexpected.

When we try to get a name of the exception _type_ we return its _message_?? I don't get it. Why?

Reply:
Weird, isn't it? 
Well, there is a reason for it:
- https://github.com/enso-org/enso/blob/develop/test/Tests/src/Semantic/R_Interop_Spec.enso#L144
- https://github.com/enso-org/enso/blob/develop/test/Tests/src/Semantic/R_Interop_Spec.enso#L154
That's needed to keep it that way. The question is, are those tests broken in the first place?

Reply:
Interesting!

But wait - how does `error.to_text` invoke `TypeToDisplayTextNode` instead of something along the lines of `AnyToDisplayTextNode`?

Oh now I see `AnyToDisplayTextNode` just delegates to `TypeToDisplayTextNode`?? _This_ is weird IMO. Shouldn't we have the exception special handling _there_ instead?

Reply:
Seems to me that `TypeToDisplayTextNode` is a wrong place for this special case. Can we have it in `AnyToDisplayTextNode` or somewhere similar?

Reply:
Resolved